File: table-cup-backend/src/main/java/ch/teko/tablecupbackend/model/TournamentModel.java
package ch.teko.tablecupbackend.model;

import java.util.UUID;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
public class TournamentModel {

  private final UUID uuid;
  private final String name;
  private final StatusModel status;
  private final PhaseModel phase;
  private boolean isUserAllowedToJoinTournament = false;
  private boolean isUserInTournament = false;
  private boolean isUserDisqualified = false;
}

File: table-cup-frontend/src/app/model/TournamentModel.ts
import {StatusModel} from './StatusModel';
import {PhaseModel} from './PhaseModel';

export class TournamentModel {
  uuid: string
  name: string;
  status: StatusModel;
  phase: PhaseModel;
  userAllowedToJoinTournament: boolean;
  userDisqualified: boolean;
  userInTournament: boolean

  constructor(uuid: string, name: string, status: StatusModel, phase: PhaseModel, userAllowedToJoinTournament: boolean, userDisqualified: boolean, userInTournament: boolean) {
    this.uuid = uuid;
    this.name = name;
    this.status = status;
    this.phase = phase;
    this.userAllowedToJoinTournament = userAllowedToJoinTournament;
    this.userDisqualified = userDisqualified;
    this.userInTournament = userInTournament;
  }
}
